### PR TITLE
Secure URLs (1/2)

### PR DESCRIPTION
**Pull request by vszakats**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/mdxmini.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class Mdxmini < Formula
   desc "Plays music in X68000 MDX chiptune format"
-  homepage "http://clogging.web.fc2.com/psp/"
+  homepage "https://clogging.web.fc2.com/psp/"
```
Comment (ilovezfs) [inline Formula/mdxmini.rb]: Is this one actually working? Seems to go straight back to http.

Comment (leonklingele) [inline Formula/mdxmini.rb]: Not our problem. Vendor might stay at https in the future.

Comment (vszakats) [inline Formula/mdxmini.rb]: It's valid, but it's a redirect to HTTP indeed. We used to consider these cases valid candidates for an URL upgrade. Probably something to be fixed on the webserver though, similarly to mixed-content issues.

**Comment by ilovezfs** [comment]
:closed_lock_with_key:

**Comment by MikeMcQuaid** [comment]
Do we know why the audit didn't catch any of these?

**Comment by ilovezfs** [comment]
No clue.

**Comment by vszakats** [comment]
Most of these were ETag-based matches.

Can it be that none of these formulae were updated since the protocol audit checks had been put in place?

**Comment by MikeMcQuaid** [comment]
@ilovezfs was meaning that question for @vszakats; I'd really like us to try and prioritise fixing exiting automated tooling.

@vszakats That's possible but `brew audit --strict` should have revealed them. Can you see if with this PR reverted locally they do?

**Comment by MikeMcQuaid** [comment]
`brew audit --strict --online` on a >=10.11 to be clear.

**Comment by ilovezfs** [comment]
I gave up on expecting that audit to work a while ago.

**Comment by vszakats** [comment]
Tested on `sylpheed.rb` it does detect both:
```
$ brew audit --strict --online ./sylpheed.rb 
error: could not lock config file /usr/local/Homebrew/.git/config: Permission denied
sylpheed:
  * The URL http://sylpheed.sraoss.jp/en/ could use HTTPS rather than HTTP
  * Stable: The URL http://sylpheed.sraoss.jp/sylpheed/v3.5/sylpheed-3.5.1.tar.bz2 could use HTTPS rather than HTTP
Error: 2 problems in 1 formula
```

In case of `znc.rb`, only the `url` is detected, because it's not an ETag match:
```
$ brew audit --strict --online ./znc.rb 
error: could not lock config file /usr/local/Homebrew/.git/config: Permission denied
znc:
  * Stable: The URL http://znc.in/releases/archive/znc-1.6.5.tar.gz could use HTTPS rather than HTTP
Error: 1 problem in 1 formula
```

On the other hand the PR message in `PULL_REQUEST_TEMPLATE.md` is missing the `--online` option, which may be a reason why it's not always run that way.

**Comment by ilovezfs** [comment]
```
iMac-TMP:~ joe$ for f in autopano-sift-c lemon mdxmini nicotine-plus nzbget pgpool-ii signify-osx sylpheed; do echo "$f ..."; brew audit --strict --online $f; done
autopano-sift-c ...
lemon ...
lemon:
  * The URL http://www.hwaci.com/sw/lemon/ could use HTTPS rather than HTTP
Error: 1 problem in 1 formula
mdxmini ...
mdxmini:
  * The URL http://clogging.web.fc2.com/psp/ could use HTTPS rather than HTTP
  * stable version should not decrease (from 20140608 to 1.0.0)
Error: 2 problems in 1 formula
nicotine-plus ...
nicotine-plus:
  * The URL http://www.nicotine-plus.org/ could use HTTPS rather than HTTP
Error: 1 problem in 1 formula
nzbget ...
nzbget:
  * Formula name conflicts with caskroom/cask/nzbget
  * The URL http://nzbget.net/ could use HTTPS rather than HTTP
Error: 2 problems in 1 formula
pgpool-ii ...
pgpool-ii:
  * Stable: The URL http://www.pgpool.net/download.php?f=pgpool-II-3.6.2.tar.gz could use HTTPS rather than HTTP
Error: 1 problem in 1 formula
signify-osx ...
signify-osx:
  * The URL http://man.openbsd.org/OpenBSD-current/man1/signify.1 could use HTTPS rather than HTTP
Error: 1 problem in 1 formula
sylpheed ...
sylpheed:
  * The URL http://sylpheed.sraoss.jp/en/ could use HTTPS rather than HTTP
  * Stable: The URL http://sylpheed.sraoss.jp/sylpheed/v3.5/sylpheed-3.5.1.tar.bz2 could use HTTPS rather than HTTP
Error: 2 problems in 1 formula
iMac-TMP:~ joe$ 
```

**Comment by ilovezfs** [comment]
My guess here is that these didn't have https when the audit was run on everything.

@vszakats any idea what's up with autopano-sift-c?

**Comment by MikeMcQuaid** [comment]
Good to know these are working better than expected.

> In case of znc.rb, only the url is detected, because homepage is not an ETag match:

@vszakats Do you think that's something that we should loosen our check on?

**Comment by vszakats** [comment]
@MikeMcQuaid I think the current ETag checks are perfectly fine. The ratio of non-ETag matches is pretty low overall. Of these, the complete content matches could be implemented (perhaps with the protocol stripping twist to give it some more chance). Partial/heuristic matches always need manual verification, so they are likely not the best fit for the audit.

@ilovezfs `autopano-sift-c` falls into this latter (partial/heuristic content match) category: The returned page contains an ID that is different on each request.

**Comment by MikeMcQuaid** [comment]
Ok, seems fine and my worries are removed. Thanks both!

**Pull request merged**